### PR TITLE
Remove string splits with `\r` from renderer resource parsers

### PR DESCRIPTION
**Pull request by heinezen**
Adds a new dedicated `util::split_newline` function that can handle safe splits.

### CODE REVIEW

File: libopenage/util/strings.h [hunk 1]
```diff
@@ -116,5 +116,10 @@ std::vector<std::string> split_escape(const std::string &txt,
                                       char delim,
                                       size_t size_hint = 0);
 
+/**
+ * Newline splitter that works with both \n and \r\n.
+ */
+std::vector<std::string> split_newline(const std::string &txt);
```
Comment (TheJJ) [inline libopenage/util/strings.h:122]: could be returning std::string_views instead :)

File: libopenage/renderer/resources/parser/parse_blendtable.cpp [hunk 2]
```diff
@@ -67,16 +67,16 @@ PatternData parse_pattern(const std::vector<std::string> &args) {
...
-	if (not file.is_file()) [[unlikely]] {
+	if (not path.is_file()) [[unlikely]] {
 		throw Error(MSG(err) << "Reading .bltable file '"
-		                     << file.get_name()
+		                     << path.get_name()
 		                     << "' failed. Reason: File not found");
 	}
 
-	auto content = file.open();
-	auto lines = content.get_lines();
```
Comment (TheJJ) [inline libopenage/renderer/resources/parser/parse_blendtable.cpp:79]: we should improve the get_lines function to improve this for all read files?

Comment (heinezen) [reply libopenage/renderer/resources/parser/parse_blendtable.cpp:79]: I've modified it to use the new `util::split_newline` function now.

File: libopenage/util/strings.h (hunk 1 above)
Comment (heinezen) [reply libopenage/util/strings.h:122]: I tried, but it is creating so many problems with existing code that relies on `std::string` which also makes the code harder to read :/ 

For example, `std::stoul` won't accept a string view as an argument, so you have this confusing back and forth conversion between std::string_view and std::string in the renderer parsers...

**Review by TheJJ — APPROVED** [review]

**Pull request merged**